pkg/labels/labels.go

```diff
@@ -12,8 +12,11 @@ import (
 // if you need labels to filter component then use GetSelector instead
 // Note: isPartOfComponent denotes if the label is required for a core resource(deployment, svc, pvc, pv) of a given component deployed with `odo dev`;
 // it is the only thing that sets it apart from the resources created via other ways (`odo deploy`, deploying resource with apply command during `odo dev`)
-func GetLabels(componentName string, applicationName string, mode string, isPartOfComponent bool) map[string]string {
+func GetLabels(componentName string, applicationName string, runtime string, mode string, isPartOfComponent bool) map[string]string {
 	labels := getLabels(componentName, applicationName, mode, true, isPartOfComponent)
+	if runtime != "" {
+		labels[openshiftRunTimeLabel] = runtime
+	}
```

Comment:
This is an OpenShift specific label, isn't it? It does not make sense to add it to resources that might be deployed in a K8s cluster.

Reply:
Do you suggest to add this label only when deploying to an OpenShift cluster?

Reply:
I'm not sure it is what has been discussed. @kadel WDYT?

Reply:
This is just a label. Yes, it is OpenShift specific, but there is no harm in always applying this. It might be even better, as there might be another tooling that could eventually leverage that.
And the only OpenShift specific is it's name, functionality wise there is nothing that should tie this only to openshift